adapters/tappx/tappx.go

```diff
@@ -0,0 +1,207 @@
+package tappx
+
+import (
+	"encoding/json"
+	"fmt"
+	"text/template"
+	//"strings"
+	"github.com/golang/glog"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	//"github.com/prebid/prebid-server/config"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+const TAPPX_BIDDER_VERSION = "1.0"
+
+type TappxAdapter struct {
+	http             *adapters.HTTPAdapter
+	endpointTemplate template.Template
+}
+
+func NewTappxAdapter(config *adapters.HTTPAdapterConfig, endpointTemplate string) adapters.Bidder {
+	return NewTappxBidder(adapters.NewHTTPAdapter(config).Client, endpointTemplate)
+}
+
+func NewTappxBidder(client *http.Client, endpointTemplate string) *TappxAdapter {
+	a := &adapters.HTTPAdapter{Client: client}
+	template, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		glog.Fatal("Unable to parse endpoint url template")
+		return nil
+	}
+	return &TappxAdapter{
+		http:             a,
+		endpointTemplate: *template,
+	}
+}
+
+type tappxParams struct {
+	Host     string `json:"host"`
+	TappxKey string `json:"tappxkey"`
+	Endpoint string `json:"endpoint"`
+}
+
+func (a *TappxAdapter) Name() string {
+	return "tappx"
+}
+
+func (a *TappxAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func (a *TappxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}}
+	}
+
+	errs := make([]error, 0, len(request.Imp))
+
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(request.Imp[0].Ext, &bidderExt); err != nil { // 3
+		errs = append(errs, err)
+		return nil, errs
```

Comment:
This is a minor thing but if we exit the function whenever we find an error as we do in lines 60 and 84, maybe we don't need the `errs` local variable declared in line 65 at all. We can exit the function, say in line 70 like:
```
 70         return nil, []error{err}
```